File: 1909. Remove One Element to Make the Array Strictly Increasing/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var canBeIncreasing = function(nums) {
  let found = false;
  let last = 0;
  for (let index = 0; index < nums.length; index++){
    let num = nums[index];
    if (last < num){
      last = num;
    }
    else{
      if (found){
        return false;
      }
      if (index === 1 || nums[index - 2] < num){
        last = num;
      }
      found = true;
    }
  }

  return true;
};

File: 1909. Remove One Element to Make the Array Strictly Increasing/solution.java
class Solution {
  public boolean canBeIncreasing(int[] nums) {
    boolean found = false;
    int last = 0;
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (last < num){
        last = num;
      }
      else{
        if (found){
          return false;
        }
        if (index == 1 || nums[index - 2] < num){
          last = num;
        }
        found = true;
      }
    }

    return true;
  }
}